gnat2goto/driver/tree_walk.adb

```diff
@@ -4531,6 +4531,28 @@ package body Tree_Walk is
    --------------------------
 
    procedure Process_Declaration (N : Node_Id; Block : Irep) is
+      procedure Handle_Representation_Clause (N : Node_Id);
+      procedure Handle_Representation_Clause (N : Node_Id) is
+         Attr_Id : constant String := Get_Name_String (Chars (N));
+         Target_Name : constant Irep := Do_Identifier (Name (N));
+         Target_Type : constant Irep :=
+           Follow_Symbol_Type (Get_Type (Target_Name), Global_Symbol_Table);
+      begin
+         if Kind (Target_Type) in Class_Type then
+            if Attr_Id = "size" then
+               if Get_Width (Target_Type) >=
+                 Integer (UI_To_Int (Intval (Expression (N))))
+               then
+                  --  the type is large enough to store values of the size
```

Comment:
I'm not sure that it is an optimisation.  It matters for things like unchecked conversion.  @tjj2017 can you give some commentary on this?  I have yet to find in the ARM exactly what the rules for `'Size' are.  One key piece of implementation advice from the standard is:

"A Size clause should be supported for an object if the specified Size is at least as large as its
subtype's Size, and corresponds to a size in storage elements that is a multiple of the object's
Alignment (if the Alignment is nonzero)."

Reply:
At least for `size` the front-end seems to present the gnat2goto with already altered types. I changed the commit from ignoring to checking that the size specified by the `size` clause is the same as the one the type has in the symbol table. `Component_size` may require more careful handling but I think this is sufficient for `size`.

Reply:
OK; that sounds like a good thing to do.